flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/AggregateAggFunction.scala

```diff
@@ -18,69 +18,51 @@
 
 package org.apache.flink.table.runtime.aggregate
 
-import java.util.{ArrayList => JArrayList, List => JList}
-import org.apache.flink.api.common.functions.{AggregateFunction => DataStreamAggFunc}
-import org.apache.flink.table.functions.{Accumulator, AggregateFunction}
+import org.apache.flink.api.common.functions.AggregateFunction
+import org.apache.flink.table.codegen.{Compiler, GeneratedAggregationsFunction}
 import org.apache.flink.types.Row
+import org.slf4j.LoggerFactory
 
 /**
   * Aggregate Function used for the aggregate operator in
   * [[org.apache.flink.streaming.api.datastream.WindowedStream]]
   *
-  * @param aggregates       the list of all [[org.apache.flink.table.functions.AggregateFunction]]
-  *                         used for this aggregation
-  * @param aggFields   the position (in the input Row) of the input value for each aggregate
+  * @param genAggregations Generated aggregate helper function
   */
 class AggregateAggFunction(
-    private val aggregates: Array[AggregateFunction[_]],
-    private val aggFields: Array[Array[Int]])
-  extends DataStreamAggFunc[Row, Row, Row] {
+    genAggregations: GeneratedAggregationsFunction)
+  extends AggregateFunction[Row, Row, Row]
```

Comment:
Hmm, OK...
I think we have to check that `function` is initialized in all methods, because `createAccumulator()` might not be called in all cases (e.g., on recovery or starting from a savepoint).